widget/view.controller.js

```diff
@@ -48,23 +48,86 @@
       'finishPageImage': widgetBasePath + 'images/ioc-extraction-finish-both.png'
     };
 
-    // Functions
-    $scope._buildPayload = _buildPayload;
+    // Wizard Functions
     $scope.moveNext = moveNext;
     $scope.moveBack = moveBack;
+
+    // Support Functions
+    $scope._getRegexPattern = _getRegexPattern;
+    $scope._buildPayload = _buildPayload;
+
+    // Exclusion List Functions
     $scope.commitExclusionSettings = commitExclusionSettings;
     $scope.validateIOC = validateIOC;
+
+    // Search Functions
     $scope.setSearchStatus = setSearchStatus;
     $scope.updateSearchQuery = updateSearchQuery;
-    $scope._getRegexPattern = _getRegexPattern;
+
+    // Bulk Import Functions
     $scope.uploadFiles = uploadFiles;
     $scope.setBulkImportFlags = setBulkImportFlags;
+    $scope._extractIOCsFromFile = _extractIOCsFromFile;
+    $scope.importIOCsFromFile = importIOCsFromFile;
+
+    // Add New Indicator Type Functions
     $scope.setAddNewIOCFlags = setAddNewIOCFlags;
     $scope.getNotEnteredIOCTypes = getNotEnteredIOCTypes;
     $scope.indicatorTypeChanged = indicatorTypeChanged;
     $scope.saveNewIOCType = saveNewIOCType;
 
 
+    function importIOCsFromFile() {
+      Object.entries(bulkImportIOCs).forEach(function ([iocType, iocList]) {
+        if (Array.isArray(iocList) && iocList.length > 0 && !ignoredIndicatorTypes.includes(iocType)) {
+          let mapping = iocExtractionConfigService.constants().iocTypeNameMapping;
+          let indicatorType = iocType;
+          for (const [key, value] of Object.entries(mapping)) {
+            if (value.includes(iocType)) {
+              indicatorType = key;
+              break;
+            }
+          }
+
+          if ($scope.updatedExclusionSettings.recordValue[indicatorType]) {
+            $scope.updatedExclusionSettings.recordValue[indicatorType].excludedIOCs = Array.from(
+              new Set([...$scope.updatedExclusionSettings.recordValue[indicatorType].excludedIOCs, ...iocList])
+            );
+          }
+
+        }
+      });
+      setBulkImportFlags('bulkImportDisable');
+      toaster.success({ body: $scope.viewWidgetVars.EXCLUDELIST_CONFIG_PAGE_BULK_IMPORT_COMPLETED });
+    }
+
+
+    function _extractIOCsFromFile(fileIRI) {
+      iocExtractionConfigService.getFileContent(fileIRI).then(function (fileContent) {
+        if (!fileContent || !fileContent.data || !fileContent.data.extracted_text) {
+          toaster.error({ body: $scope.viewWidgetVars.EXCLUDELIST_CONFIG_PAGE_BULK_IMPORT_FILE_CONTENT_INVALID });
```

Comment:
You can use shorter names as well here, e.g.   EXCLUDELIST_CONFIG_PAGE_BULK_IMPORT_FILE_CONTENT_INVALID ---> BULK_IMPORT_INVALID_FILE_CONTENT_ERR_MSG

EXCLUDELIST_CONFIG_PAGE_BULK_IMPORT_COMPLETED --> BULK_IMPORT_COMPLETED_SUCCESS_MSG

Reply:
@cs-suyog-jinturkar, I use a naming convention like <PAGE_NAME>_<VARIABLE_NAME> to easily identify which variable belongs to which page in the wizard, also in HTML, controller.js, and i18n files. Because of this, I prefer not to change the variable names.